x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/analyses/DataFrameAnalysis.java

```diff
@@ -32,6 +33,13 @@ public String toString() {
 
     Type getType();
 
+    /**
+     * The fields that will contain the results of the analysis
+     *
+     * @return Set of Strings representing the result fields for the constructed analysis
+     */
+    Set<String> getResultFields();
```

Comment:
I do think a better solution here is to have some known prefix for our result fields and which analyzer they satisfy. 

Something like `_ml_analysis_result.<analyzer>.<result-name>`. But, that change is more substantial, opted for better planning and refactoring down the line when/if we make that change.